org.eclipse.scanning.event.ui/src/org/eclipse/scanning/event/ui/view/StatusQueueView.java

```diff
@@ -214,54 +195,40 @@ protected void updateSelected() {
 	/**
 	 * Listens to a topic
 	 */
-	private void createTopicListener(final URI uri) throws Exception {
+	private void createTopicListener(final URI uri) {
 
 		// Use job because connection might timeout.
 		final Job topicJob = new Job("Create topic listener") {
-
 			@Override
 			protected IStatus run(IProgressMonitor monitor) {
 				try {
 					topicMonitor = service.createSubscriber(uri, getTopicName());
-					topicMonitor.addListener(new IBeanListener<StatusBean>() {
-						@Override
-						public void beanChangePerformed(BeanEvent<StatusBean> evt) {
+					topicMonitor.addListener(evt -> {
 							final StatusBean bean = evt.getBean();
-	                        try {
+							try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)